src/uu/env/src/env.rs

```diff
@@ -256,7 +256,31 @@ fn run_env(args: impl uucore::Args) -> UResult<()> {
 
     // set specified env vars
     for &(name, val) in &opts.sets {
-        // FIXME: set_var() panics if name is an empty string
+        /*
+         * set_var panics if name is an empty string
+         * set_var internally calls setenv (on unix at least), while GNU env calls putenv instead.
+         *
+         * putenv returns successfully if provided with something like "=a" and modifies the environ
+         * variable to contain "=a" inside it, effectively modifying the process' current environment
+         * to contain a malformed string in it. Using GNU's implementation, the command `env =a`
+         * prints out the malformed string and even invokes the child process with that environment.
+         * This can be seen by using `env -i =a env` or `env -i =a cat /proc/self/environ`
+         *
+         * POSIX.1-2017 doesn't seem to mention what to do if the string is malformed (at least
+         * not in "Chapter 8, Environment Variables" or in the definition for environ and various
+         * exec*'s or in the description of env in the "Shell & Utilities" volume).
+         *
+         * It also doesn't specify any checks for putenv before modifying the environ variable, which
+         * is likely why glibc doesn't do so. However, setenv's first argument cannot point to
+         * an empty string or a string containing '='.
+         *
+         * There is no benefit in replicating GNU's env behavior, since it will only modify the
+         * environment in weird ways
+         */
+
+        if name.is_empty() {
+            continue;
```

Comment:
Maybe we could print a warning instead of silently ignoring the argument? It's most likely an unintentional mistake by the user to specify an empty variable name.

Reply:
GNU env doesn't do that because it uses putenv so I didn't add it. There doesn't seem to be any GNU tests that check this either. Should I add a verbose argument (ensuring compatibility) or is it uncessary?
Will log something like "no name specified for value {}".

Reply:
I wouldn't add a verbose argument because I don't expect this to be a compatibility problem.